presto-main/src/test/java/com/facebook/presto/cost/TestJoinStatsRule.java

```diff
@@ -0,0 +1,218 @@
+    private static final double RIGHT_JOIN_COLUMN_NON_NULLS = 0.4;
+    private static final int RIGHT_JOIN_COLUMN_NDV = 15;
+
+    private static final SymbolStatistics LEFT_OTHER_COLUMN_STATS =
+            symbolStatistics(LEFT_OTHER_COLUMN, 42, 42, 0.42, 42);
+    private static final SymbolStatistics RIGHT_OTHER_COLUMN_STATS =
+            symbolStatistics(RIGHT_OTHER_COLUMN, 24, 24, 0.24, 24);
+    private static final PlanNodeStatsEstimate LEFT_STATS = planNodeStats(LEFT_ROWS_COUNT,
+            symbolStatistics(LEFT_JOIN_COLUMN, 0.0, 10.0, LEFT_JOIN_COLUMN_NULLS, LEFT_JOIN_COLUMN_NDV),
+            LEFT_OTHER_COLUMN_STATS);
+    private static final PlanNodeStatsEstimate RIGHT_STATS = planNodeStats(RIGHT_ROWS_COUNT,
+            symbolStatistics(RIGHT_JOIN_COLUMN, 5.0, 10.0, RIGHT_JOIN_COLUMN_NULLS, RIGHT_JOIN_COLUMN_NDV),
+            RIGHT_OTHER_COLUMN_STATS);
+
+    private static final JoinStatsRule JOIN_STATS_RULE = new JoinStatsRule(new FilterStatsCalculator(createTestMetadataManager()));
+
+    private StatsCalculatorTester tester;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        tester = new StatsCalculatorTester();
+    }
+
+    @Test
+    public void testStatsForInnerJoin()
+            throws Exception
+    {
+        double innerJoinRowCount = LEFT_ROWS_COUNT * RIGHT_ROWS_COUNT / LEFT_JOIN_COLUMN_NDV * LEFT_JOIN_COLUMN_NON_NULLS * RIGHT_JOIN_COLUMN_NON_NULLS;
+        PlanNodeStatsEstimate innerJoinStats = planNodeStats(innerJoinRowCount,
+                symbolStatistics(LEFT_JOIN_COLUMN, 0.0, 10.0, 0.0, RIGHT_JOIN_COLUMN_NDV),
+                symbolStatistics(RIGHT_JOIN_COLUMN, 5.0, 10.0, 0.0, RIGHT_JOIN_COLUMN_NDV),
+                LEFT_OTHER_COLUMN_STATS, RIGHT_OTHER_COLUMN_STATS);
+
+        assertJoinStats(INNER, LEFT_STATS, RIGHT_STATS, innerJoinStats);
+    }
+
+    @Test
+    public void testStatsForLeftAntiJoin()
+    {
+        PlanNodeStatsEstimate antiJoinStats = planNodeStats(LEFT_ROWS_COUNT * (LEFT_JOIN_COLUMN_NULLS + LEFT_JOIN_COLUMN_NON_NULLS / 4),
+                symbolStatistics(LEFT_JOIN_COLUMN, 0.0, 10.0, LEFT_JOIN_COLUMN_NULLS / (LEFT_JOIN_COLUMN_NULLS + LEFT_JOIN_COLUMN_NON_NULLS / 4), 5),
+                LEFT_OTHER_COLUMN_STATS);
+
+        PlanNodeStatsAssertion.assertThat(antiJoinStats).equalTo(JOIN_STATS_RULE.calculateAntiJoinStats(
+                Optional.empty(),
+                ImmutableList.of(new JoinNode.EquiJoinClause(new Symbol(LEFT_JOIN_COLUMN), new Symbol(RIGHT_JOIN_COLUMN))),
+                LEFT_STATS, RIGHT_STATS));
+    }
+
+    @Test
+    public void testStatsForRightAntiJoin()
+    {
+        PlanNodeStatsEstimate antiJoinStats = planNodeStats(RIGHT_ROWS_COUNT * RIGHT_JOIN_COLUMN_NULLS,
+                symbolStatistics(RIGHT_JOIN_COLUMN, 5.0, 10.0, 1.0, 0),
+                RIGHT_OTHER_COLUMN_STATS);
+
+        PlanNodeStatsAssertion.assertThat(antiJoinStats).equalTo(JOIN_STATS_RULE.calculateAntiJoinStats(
```

Comment:
static import

also you flipped the convention. You should do `assertThat(actual).isEqualtTo(expected);`